gazelle_haskell_modules/lang.go

```diff
@@ -112,9 +92,15 @@ func (*gazelleHaskellModulesLang) Imports(c *config.Config, r *rule.Rule, f *rul
 	if r.Kind() == "haskell_module" {
 		originatingRules := getOriginatingRules(r)
 		moduleSpecs := make([]resolve.ImportSpec, len(originatingRules), 2*len(originatingRules)+1)
+		// When studying a 'haskell_module' rule,
+		// a field "filepath:pkg:component:relativePath" for each rule using this module is added to the RuleIndex.
+		// This field is used by 'SetNonHaskellModuleDeps' to list the modules in the libraries/binaries/tests.
```

Comment:
Some edit suggestions.
```suggestion
		// These entries are used by 'SetNonHaskellModuleDeps' to list the modules in the libraries/binaries/tests.
```
It is not necessary to duplicate in comments the syntax of the index key since the code always uses `moduleByFilepathSpec` to build the key.

`// When studying a 'haskell_module' rule` is implied by the guarding if statement and redundant.

Is it `fields` the word to use here? I think entries or records would be the usual term.

Similar criteria would apply to the other comments.

---------------

I don't oppose adding these comments, but I'd find more useful and less prone to outdate to document the meaning of each key instead of what they are used for and where. For instance:

```
// haskell_modules with a given filepath (relative to the package?)
func moduleByFilepathSpec(...) {
}

// haskell_module rules with a given module name in a given enclosing library/binary/test.
func moduleByModuleImportSpec(...) {
}

...
```

Reply:
I have the feeling that the "Why" is the question which would be the most useful to answer in the comments. Especially having a central place where we learn what we are storing and why was what missed me the most when working with the codebase.
I do not think that adding a comment "haskell_modules with a given filepath" the line above a function named `moduleByFilepath...` would help to understand the rationale of this function.